ci/docker/wasm32-wasip2/Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM ubuntu:24.04
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    ca-certificates \
+    curl \
+    clang \
+    xz-utils
+
+# Wasmtime is used to execute tests and wasi-sdk is used to compile tests.
+# Download appropriate versions here and configure various flags below.
+#
+# Note that these versions are kept in sync with the `wasm32-wasip2/Dockerfile`
+# as well.
+ENV WASMTIME 24.0.0
+ENV WASI_SDK 24
+
+RUN curl -L https://github.com/bytecodealliance/wasmtime/releases/download/v$WASMTIME/wasmtime-v$WASMTIME-x86_64-linux.tar.xz | \
+  tar xJf -
+ENV PATH=$PATH:/wasmtime-v$WASMTIME-x86_64-linux
+
+RUN curl -LO https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$WASI_SDK/wasi-sdk-$WASI_SDK.0-x86_64-linux.deb
+RUN dpkg -i ./wasi-sdk-*.deb
+
+# Note that most of these are copied from `wasm32-wasip1/Dockerfile`
+#
+# FIXME: the `-Clink-arg` to export `cabi_realloc` is a bug in the target
+# itself, this should be fixed upstream.
+ENV CARGO_TARGET_WASM32_WASIP2_RUNNER=wasmtime \
+    CARGO_TARGET_WASM32_WASIP2_LINKER=/opt/wasi-sdk/bin/clang \
+    CARGO_TARGET_WASM32_WASIP2_RUSTFLAGS="-lwasi-emulated-process-clocks -Ctarget-feature=-crt-static -Clink-arg=-Wl,--export,cabi_realloc" \
+    CC_wasm32_wasip2=/opt/wasi-sdk/bin/clang \
+    CFLAGS_wasm32_wasip2=-D_WASI_EMULATED_PROCESS_CLOCKS \
+    PATH=$PATH:/rust/bin
```

Comment:
Since most of this file is the same as wasip1, could it move to a script? E.g. we have `ci/emscripten.sh` and a couple others that are used in multiple dockerfiles. Otherwise lgtm.

Reply:
Good idea! I've extracted this out now